### PR TITLE
Add openxlsx2.na.strings option

### PR DESCRIPTION
**Pull request by olivroy**
I was thinking about something more like this.

I strongly feel about not having an option overriding a parameter passed (other than the default)

If option is set and na.string is  not supplied: use the option
If option is set and na.string is supplied: use na.strings
If option is not set and na.string is not use default.

Amends to #473 and addresses #20

### CODE REVIEW

File: R/write.R [hunk 1]
```diff
@@ -782,6 +782,12 @@ write_data_table <- function(
   if (is.null(x)) {
     return(wb)
   }
+  # overwrite na.strings if nothing was provided
+  # with whatever is in the option if not set to default
+  if (identical(as.character(na.strings), "na_string") &&
```
Comment (JanMarvin) [inline R/write.R:787]: I don’t recall at the moment if we could check for inherits waiver.

**Comment by JanMarvin** [comment]
I see why you want it this way. Since I have no preference, I’m equally happy with this PR!

File: R/write.R (hunk 1 above)
Comment (olivroy) [reply R/write.R:790]: there is `is_na_strings()`, but I am somewhat scared of it.

File: R/write.R [hunk 2]
```diff
@@ -782,6 +785,12 @@ write_data_table <- function(
...
+      !is.null(getOption("openxlsx2.na.strings"))) {
```
Comment (olivroy) [inline R/write.R:791]:
```suggestion
  if (is_na_strings(na.strings) && !is.null(getOption("openxlsx2.na.strings"))) {

```

File: R/write.R [hunk 3]
```diff
@@ -782,6 +785,12 @@ write_data_table <- function(
   if (is.null(x)) {
     return(wb)
   }
+  # overwrite na.strings if nothing was provided
+  # with whatever is in the option if not set to default
+  if (is_na_strings(na.strings) && !is.null(getOption("openxlsx2.na.strings"))) {
+
```
Comment (olivroy) [inline R/write.R:791]:
```suggestion
```

**Comment by JanMarvin** [comment]
Looks good to me! Thanks

**Pull request merged**